### ISSUE TITLE
Could not find a resource of type 'script' named 'vendor-jquery' with version 'any.

### ISSUE THREAD

**Issue by randaratceridian** [OrchardCMS/OrchardCore#4833]
When doing an initial setup of a site, I am getting _Could not find a resource of type 'script' named 'vendor-jquery' with version 'any._

    Recipe: Agency
    Database: SQL Server
        Connection string: server=localhost;Database=OrchardCorePOC2;User Id=OrchardCorePOC2;Password=OrchardCorePOC2
    Orchard Core: 1.0.0-rc1
    Dotnet core version: 3.0.100
    SQL Server: 13.0.5233.0

It looks like the tables are being created in SQL server.
`select count(*) from sys.tables;`
Returns 13 rows

Full stack trace here:

```
System.InvalidOperationException: Could not find a resource of type 'script' named 'vendor-jquery' with version 'any.
   at OrchardCore.ResourceManagement.ResourceManager.DoGetRequiredResources(String resourceType) in C:\projects\orchardcore\src\OrchardCore\OrchardCore.ResourceManagement\ResourceManager.cs:line 370
   at OrchardCore.ResourceManagement.ResourceManager.RenderHeadScript(IHtmlContentBuilder builder) in C:\projects\orchardcore\src\OrchardCore\OrchardCore.ResourceManagement\ResourceManager.cs:line 566
   at OrchardCore.ResourceManagement.TagHelpers.ResourcesTagHelper.Process(TagHelperContext tagHelperContext, TagHelperOutput output) in C:\projects\orchardcore\src\OrchardCore\OrchardCore.ResourceManagement\TagHelpers\ResourcesTagHelper.cs:line 47
   at Microsoft.AspNetCore.Razor.TagHelpers.TagHelper.ProcessAsync(TagHelperContext context, TagHelperOutput output)
   at OrchardCore.DisplayManagement.Liquid.Tags.HelperStatement.WriteToAsync(TextWriter writer, TextEncoder encoder, TemplateContext context) in C:\projects\orchardcore\src\OrchardCore\OrchardCore.DisplayManagement.Liquid\Tags\HelperTag.cs:line 121
   at Fluid.BaseFluidTemplate`1.Awaited(ValueTask`1 task, TextWriter writer, TextEncoder encoder, TemplateContext context, List`1 statements, Int32 startIndex)
   at OrchardCore.DisplayManagement.Liquid.LiquidViewTemplateExtensions.RenderAsync(LiquidViewTemplate template, LiquidOptions options, IServiceProvider services, TextWriter writer, TextEncoder encoder, TemplateContext templateContext) in C:\projects\orchardcore\src\OrchardCore\OrchardCore.DisplayManagement.Liquid\LiquidViewTemplate.cs:line 196
   at OrchardCore.DisplayManagement.Liquid.LiquidViewTemplate.RenderAsync(RazorPage`1 page) in C:\projects\orchardcore\src\OrchardCore\OrchardCore.DisplayManagement.Liquid\LiquidViewTemplate.cs:line 115
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderPageCoreAsync(IRazorPage page, ViewContext context)
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderPageAsync(IRazorPage page, ViewContext context, Boolean invokeViewStarts)
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderAsync(ViewContext context)
   at Microsoft.AspNetCore.Mvc.ViewFeatures.HtmlHelper.RenderPartialCoreAsync(String partialViewName, Object model, ViewDataDictionary viewData, TextWriter writer)
   at Microsoft.AspNetCore.Mvc.ViewFeatures.HtmlHelper.PartialAsync(String partialViewName, Object model, ViewDataDictionary viewData)
   at OrchardCore.DisplayManagement.Implementation.DefaultHtmlDisplay.ProcessAsync(ShapeBinding shapeBinding, IShape shape, DisplayContext context) in C:\projects\orchardcore\src\OrchardCore\OrchardCore.DisplayManagement\Implementation\DefaultHtmlDisplay.cs:line 268
   at OrchardCore.DisplayManagement.Implementation.DefaultHtmlDisplay.ExecuteAsync(DisplayContext context) in C:\projects\orchardcore\src\OrchardCore\OrchardCore.DisplayManagement\Implementation\DefaultHtmlDisplay.cs:line 119
   at OrchardCore.DisplayManagement.Implementation.DefaultHtmlDisplay.ExecuteAsync(DisplayContext context) in C:\projects\orchardcore\src\OrchardCore\OrchardCore.DisplayManagement\Implementation\DefaultHtmlDisplay.cs:line 173
   at CallSite.Target(Closure , CallSite , Object )
   at System.Dynamic.UpdateDelegates.UpdateAndExecute1[T0,TRet](CallSite site, T0 arg0)
   at OrchardCore.DisplayManagement.Theming.ThemeLayout.ExecuteAsync() in C:\projects\orchardcore\src\OrchardCore\OrchardCore.DisplayManagement\Theming\ThemeLayout.cs:line 26
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderPageCoreAsync(IRazorPage page, ViewContext context)
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderPageAsync(IRazorPage page, ViewContext context, Boolean invokeViewStarts)
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderLayoutAsync(ViewContext context, ViewBufferTextWriter bodyWriter)
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderAsync(ViewContext context)
   at Microsoft.AspNetCore.Mvc.ViewFeatures.ViewExecutor.ExecuteAsync(ViewContext viewContext, String contentType, Nullable`1 statusCode)
   at Microsoft.AspNetCore.Mvc.ViewFeatures.ViewExecutor.ExecuteAsync(ActionContext actionContext, IView view, ViewDataDictionary viewData, ITempDataDictionary tempData, String contentType, Nullable`1 statusCode)
   at Microsoft.AspNetCore.Mvc.ViewFeatures.ViewResultExecutor.ExecuteAsync(ActionContext context, ViewResult result)
   at Microsoft.AspNetCore.Mvc.ViewResult.ExecuteResultAsync(ActionContext context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResultFilterAsync>g__Awaited|29_0[TFilter,TFilterAsync](ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResultExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.ResultNext[TFilter,TFilterAsync](State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeResultFilters>g__Awaited|27_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|24_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at SixLabors.ImageSharp.Web.Middleware.ImageSharpMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware.<Invoke>g__AwaitMatch|8_1(EndpointRoutingMiddleware middleware, HttpContext httpContext, Task matchTask)
   at OrchardCore.Diagnostics.DiagnosticsStartupFilter.<>c__DisplayClass3_0.<<Configure>b__1>d.MoveNext() in C:\projects\orchardcore\src\OrchardCore.Modules\OrchardCore.Diagnostics\DiagnosticsStartupFilter.cs:line 36
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Diagnostics.StatusCodePagesMiddleware.Invoke(HttpContext context)
   at OrchardCore.Modules.ModularTenantRouterMiddleware.Invoke(HttpContext httpContext) in C:\projects\orchardcore\src\OrchardCore\OrchardCore\Modules\ModularTenantRouterMiddleware.cs:line 82
   at OrchardCore.Environment.Shell.Scope.ShellScope.UsingAsync(Func`2 execute) in C:\projects\orchardcore\src\OrchardCore\OrchardCore.Abstractions\Shell\Scope\ShellScope.cs:line 101
   at OrchardCore.Modules.ModularTenantContainerMiddleware.Invoke(HttpContext httpContext) in C:\projects\orchardcore\src\OrchardCore\OrchardCore\Modules\ModularTenantContainerMiddleware.cs:line 59
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
```

**Comment by randaratceridian** [comment]
Unfortunately, I am no closer to figuring this out. I followed the same steps on a different computer and it installed with no issue. The difference I can find are:

- The failing one is my work computer, so is generally more locked down. But I am a local admin and this is .Net core, so I'm not clear how any policies could affect this.
- The working computer is using SQL 15 while the failing one is using SQL 13. 

The steps I followed were:

1. dotnet new -i OrchardCore.ProjectTemplates::1.0.0-*
2. dotnet new occms -n OrchardCorePOC
3. dotnet run --project OrchardCorePOC.csproj
4. Browse to https://localhost:5001/
5. Setup the site with valid database settings

The only thing I’ve found odd is the tables being different when I run this query - `select name from sys.tables order by name`

**Working version**
ContentItemIndex
DeploymentPlanIndex
Document
Identifiers
UserByClaimIndex
UserByLoginInfoIndex
UserByRoleNameIndex
UserByRoleNameIndex_Document
UserIndex

**Failing version**
AliasPartIndex
AutoroutePartIndex
ContainedPartIndex
ContentItemIndex
DeploymentPlanIndex
Document
Identifiers
LayerMetadataIndex
UserByClaimIndex
UserByLoginInfoIndex
UserByRoleNameIndex
UserByRoleNameIndex_Document
UserIndex

Since the problem seems to be related to resources, I’ve been trying to understand more about how resources are loaded. One thing I would like to try is UseCDN from https://orchardcore.readthedocs.io/en/dev/docs/reference/modules/Resources/#usecdn. But I’m not clear where that is set. Is that something I can add to appsettings.json?

**Comment by deanmarcussen** [comment]
This is confusing.

SQL Server 13 is sql server 2016 , which is one of the versions I have run fine with Orchard Core.

But your failed version table listing looks very different to your working version listing.
Are you sure you ran the same recipe on both? The working version looks much more like the saas recipe (but not quite, which is strange) than the agency recipe.

However I would recommend you try it with sqlite. sqlite is super easy, and really well suited to dev purposes. particular if your machine is locked down by work.

This will help you isolate your issue as to whether it is database / setup related, or something else.

I can't repro it, so I'm suspecting a setup database issue, maybe a nuget cache issue.

The taghelper which is trying to load that resource is loading it from a `ResourceManifest` class in the agency theme. The manifest is loaded from `Startup` in the theme. That manifest contains the cdn url's, but from your error it looks like `Startup` hasn't been called, so the script is unregistered. 

Another thing to try, because you said you tried it with RC1, but then you are maybe using the myget feed (you must be, because this script/feature is post RC1), is to try a dotnet clean and rebuild. If you have moved from RC1, to the dev feed, some people have experienced problems with the nuget cache not being updated correctly, and old assemblies being around.

When you do have the site running the UseCdn setting is controlled via the general site settings page in the admin `/admin`

**Comment by randaratceridian** [comment]
Thank you Dean for your response. Unfortunately, I'm still banging my head against this issue but have more information

I tried following the same steps on a co-workers desktop and it worked with no issue, the first time. In theory, his machine should be as locked down as mine. So that eliminates some weird security configuration.

I also did a full purge of everything Orchard in the hopes it would fix the issue.

1.	Deleted all of my folders generated with dotnet new. I had a bunch from all my attempts but wanted a clean slate.
2.	Deleted all of the databases
3.	Re-installed the templates

```
dotnet new -u OrchardCore.ProjectTemplates
dotnet new -i OrchardCore.ProjectTemplates::1.0.0-*

```
4.	Re-created the site. I’m just using the Agency recipe for now since that is the one that will demo best.

Now the key point. I created it using **_SQLite_**. Unfortunately, I got the exact same error. `InvalidOperationException: Could not find a resource of type 'script' named 'vendor-jquery' with version 'any.` So at least it narrows it down to being a code issue, not a database issue.

So, unfortunately, I’m back to square one.

I know I said the DB was not the problem, but to answer your question, it does look like the table discrepancy was because of the different recipes. I assumed they would create the same tables for any recipe, but I guess not. The list of tables created is:

- AliasPartIndex
- AutoroutePartIndex
- ContainedPartIndex
- ContentItemIndex
- DeploymentPlanIndex
- Document
- Identifiers
- LayerMetadataIndex
- UserByClaimIndex
- UserByLoginInfoIndex
- UserByRoleNameIndex
- UserByRoleNameIndex_Document
- UserIndex

**Comment by randaratceridian** [comment]
The good news is I’ve made some more progress, but not at the point that I want to be.
I pulled down the dev branch, built it and ran the OrchardCore.Cms.Web project the same way I have the sites created from the templates. i.e. `dotnet run --project OrchardCore.Cms.Web.csproj`. The site ran just fine, even with SQL server! So, on a positive note:

- The theory is that once OrchardCore is out of RC and dev becomes the new master, the problem may go away.
- I can get through my POC to evaluate if Orchard Core is what we are looking for.

On a negative note, I have no idea why it works everywhere except the one computer it really needs to run on. I can’t keep using the dev branch if we decide to proceed with Orchard Core. So hoping the plan is to release 1.0.0 soon.

**Comment by deanmarcussen** [comment]
That's great you've made some progress.

Somehow I think you've ended up with a copy of the myget feed in your nuget cache.

Nuget cache is located in `C:\Users\dean\.nuget\packages` or similar.

The reason i say this is because the feature for registered scripts in the agency theme was not in RC1, but is available on the dev myget feed.

Here's a link to the codegen docs
https://orchardcore.readthedocs.io/en/dev/docs/getting-started/templates/
because
> dotnet new -u OrchardCore.ProjectTemplates
> dotnet new -i OrchardCore.ProjectTemplates::1.0.0-*

doesn't make sense to me. it means it may grab the myget version **if it has been cached previously**

So suggestion is
- either clear your nuget cache and try again
- use the dev myget feed when installing the templates

if you're still having problems, open the '.csproj` files and post the versions that have been installed, because with that it is very hard to understand which version you may be trying to use.

**Comment by randaratceridian** [comment]
Hello @sebastienros. I am not sure why this was closed. The issue is not resolved. Please re-open it.

@deanmarcussen. You said that `dotnet new -i OrchardCore.ProjectTemplates::1.0.0-*` does not make sense. This is directly from the documentation here https://orchardcore.readthedocs.io/en/dev/docs/guides/create-cms-application/. If that is incorrect, I would suggest it get updated.

Now the worst news. The problem is still there after the following:

1. Ran `dotnet new -u OrchardCore.ProjectTemplates`
1. Removed C:\Users\<myuser>\\.nuget\packages\orchardcode.*
1. Ran `dotnet new -i OrchardCore.ProjectTemplates::1.0.0-rc1-*` as per https://orchardcore.readthedocs.io/en/dev/docs/getting-started/templates/
1. Created the new site using `dotnet new occms -n OrchardCorePOC_Agency`
1. dotnet run --project OrchardCorePOC_Agency.csproj
1. Ran the setup process from https://localhost:5001/. Database tables were created, but when running the site, I still get:
`InvalidOperationException: Could not find a resource of type 'script' named 'vendor-jquery' with version 'any.`

This is what is in OrchardCorePOC_Agency.csproj

```
  <ItemGroup>
    <PackageReference Include="OrchardCore.Logging.NLog" Version="1.0.0-rc1-10004" />
    <PackageReference Include="OrchardCore.Application.Cms.Targets" Version="1.0.0-rc1-10004" />
  </ItemGroup>
```

Trying something different, I tried to use the latest dev version.

1. Ran `dotnet new -u OrchardCore.ProjectTemplates`
1. Removed C:\Users\<myuser>\\.nuget\packages\orchardcode.*
1. Ran `dotnet new -i OrchardCore.ProjectTemplates::1.0.0-rc1-* --nuget-source https://www.myget.org/F/orchardcore-preview/api/v3/index.json`
1. Deleted my old site
1. Created the new site using `dotnet new occms -n OrchardCorePOC_Agency`
1. dotnet run --project OrchardCorePOC_Agency.csproj

Unfortunately, it seems to have an issue getting a package with the latest dev version

```
C:\Projects\OrchardCorePOC_Agency\OrchardCorePOC_Agency.csproj : error NU1102: Unable to find package OrchardCore.Application.Cms.Targets with version (>= 1.0.0-rc1-10802)
C:\Projects\OrchardCorePOC_Agency\OrchardCorePOC_Agency.csproj : error NU1102:   - Found 10 version(s) in nuget.org [ Nearest version: 1.0.0-rc1-10004 ]
C:\Projects\OrchardCorePOC_Agency\OrchardCorePOC_Agency.csproj : error NU1102:   - Found 0 version(s) in Microsoft Visual Studio Offline Packages
C:\Projects\OrchardCorePOC_Agency\OrchardCorePOC_Agency.csproj : error NU1102: Unable to find package OrchardCore.Logging.NLog with version (>= 1.0.0-rc1-10802)
C:\Projects\OrchardCorePOC_Agency\OrchardCorePOC_Agency.csproj : error NU1102:   - Found 11 version(s) in nuget.org [ Nearest version: 1.0.0-rc1-10004 ]
C:\Projects\OrchardCorePOC_Agency\OrchardCorePOC_Agency.csproj : error NU1102:   - Found 0 version(s) in Microsoft Visual Studio Offline Packages
```

This is now what is in OrchardCorePOC_Agency.csproj

```
  <ItemGroup>
    <PackageReference Include="OrchardCore.Logging.NLog" Version="1.0.0-rc1-10802" />
    <PackageReference Include="OrchardCore.Application.Cms.Targets" Version="1.0.0-rc1-10802" />
  </ItemGroup>
```

So I still can't create a working site from a template.

**Comment by Skrypt** [comment]
Ok please follow this guideline because there is a lot of commands I saw that are not like what is in the documentation. You seem to be trying to build from the master branch templates or older templates.

https://orchardcore.readthedocs.io/en/dev/docs/getting-started/templates/

**Comment by Skrypt** [comment]
Have you tried with :

`dotnet new -i OrchardCore.ProjectTemplates::1.0.0-rc1-* --nuget-source https://www.myget.org/F/orchardcore-preview/api/v3/index.json  `

**Comment by Skrypt** [comment]
Ok tried this morning and in fact there is an issue with the templates. Here is what I did.

```
dotnet new -i OrchardCore.ProjectTemplates::1.0.0-rc1-* --nuget-source https://www.myget.org/F/orchardcore-preview/api/v3/index.json
dotnet new occms
dotnet run
```

This will fail resolving the nuget packages required to run this app.
The solution for fixing this issue temporarly is to go into the .csproj file and edit the PackageReference versions.

```
  <ItemGroup>
    <PackageReference Include="OrchardCore.Logging.NLog" Version="1.0.0-rc1-*" />
    <PackageReference Include="OrchardCore.Application.Cms.Targets" Version="1.0.0-rc1-*" />
  </ItemGroup>
```

This way will always try to retrieve the latest dev Nuget package versions.

**Comment by randaratceridian** [comment]
Hello @Skrypt. I was so happy to see somebody was looking back into this. Unfortunately, I seem to be in the same spot as before. Steps I followed:

1. nuget locals all -clear
2. dotnet new -i OrchardCore.ProjectTemplates::1.0.0-rc1-* --nuget-source https://www.myget.org/F/orchardcore-preview/api/v3/index.json
3. dotnet new occms -n OrchardCorePOC_Agency
4. Changed OrchardCorePOC_Agency.csproj

```
  <ItemGroup>
    <!--<PackageReference Include="OrchardCore.Logging.NLog" Version="1.0.0-rc1-11154" />
    <PackageReference Include="OrchardCore.Application.Cms.Targets" Version="1.0.0-rc1-11154" />-->
	<PackageReference Include="OrchardCore.Logging.NLog" Version="1.0.0-rc1-*" />
    <PackageReference Include="OrchardCore.Application.Cms.Targets" Version="1.0.0-rc1-*" />
  </ItemGroup>
```

Ran the setup as usual using the Agency recipe. As soon as the wizard completes, I get the exception at the bottom. I also tried:

- Using the SaaS recipe. Very similar behaviour, but `InvalidOperationException: Could not find a resource of type 'stylesheet' named 'bootstrap-oc' with version '1.`
- Tried using Nuget.config you specified.

Error with Agency recipe
```
InvalidOperationException: Could not find a resource of type 'script' named 'vendor-jquery' with version 'any'.
OrchardCore.ResourceManagement.ResourceManager.DoGetRequiredResources(string resourceType) in ResourceManager.cs
+
                    throw new InvalidOperationException($"Could not find a resource of type '{settings.Type}' named '{settings.Name}' with version '{settings.Version ?? "any"}'.");
OrchardCore.ResourceManagement.ResourceManager.RenderHeadScript(IHtmlContentBuilder builder) in ResourceManager.cs
+
            var headScripts = DoGetRequiredResources("script");
OrchardCore.ResourceManagement.TagHelpers.ResourcesTagHelper.Process(TagHelperContext tagHelperContext, TagHelperOutput output) in ResourcesTagHelper.cs
+
                    _resourceManager.RenderHeadScript(output.Content);
Microsoft.AspNetCore.Razor.TagHelpers.TagHelper.ProcessAsync(TagHelperContext context, TagHelperOutput output)
OrchardCore.DisplayManagement.Liquid.Tags.HelperStatement.WriteToAsync(TextWriter writer, TextEncoder encoder, TemplateContext context) in HelperTag.cs
+
                => Task.FromResult(new DefaultTagHelperContent().AppendHtml(content)));
Fluid.BaseFluidTemplate<T>.Awaited(ValueTask<Completion> task, TextWriter writer, TextEncoder encoder, TemplateContext context, List<Statement> statements, int startIndex)
System.Runtime.CompilerServices.ValueTaskAwaiter.GetResult()
OrchardCore.DisplayManagement.Liquid.LiquidViewTemplate.RenderAsync(RazorPage<object> page) in LiquidViewTemplate.cs
+
Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderPageCoreAsync(IRazorPage page, ViewContext context)
Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderPageAsync(IRazorPage page, ViewContext context, bool invokeViewStarts)
Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderAsync(ViewContext context)
Microsoft.AspNetCore.Mvc.ViewFeatures.HtmlHelper.RenderPartialCoreAsync(string partialViewName, object model, ViewDataDictionary viewData, TextWriter writer)
Microsoft.AspNetCore.Mvc.ViewFeatures.HtmlHelper.PartialAsync(string partialViewName, object model, ViewDataDictionary viewData)
OrchardCore.DisplayManagement.Implementation.DefaultHtmlDisplay.<ProcessAsync>g__Awaited|12_0(Task<IHtmlContent> task) in DefaultHtmlDisplay.cs
+
            static async ValueTask<IHtmlContent> Awaited(Task<IHtmlContent> task)
OrchardCore.DisplayManagement.Implementation.DefaultHtmlDisplay.ExecuteAsync(DisplayContext context) in DefaultHtmlDisplay.cs
+
                        await shapeMetadata.ProcessingAsync.InvokeAsync(processing => processing(displayContext.Shape), _logger);
OrchardCore.DisplayManagement.Implementation.DefaultHtmlDisplay.ExecuteAsync(DisplayContext context) in DefaultHtmlDisplay.cs
+
            finally
CallSite.Target(Closure , CallSite , object )
System.Dynamic.UpdateDelegates.UpdateAndExecute1<T0, TRet>(CallSite site, T0 arg0)
OrchardCore.DisplayManagement.Theming.ThemeLayout.ExecuteAsync() in ThemeLayout.cs
+
    }
Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderPageCoreAsync(IRazorPage page, ViewContext context)
Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderPageAsync(IRazorPage page, ViewContext context, bool invokeViewStarts)
Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderLayoutAsync(ViewContext context, ViewBufferTextWriter bodyWriter)
Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderAsync(ViewContext context)
Microsoft.AspNetCore.Mvc.ViewFeatures.ViewExecutor.ExecuteAsync(ViewContext viewContext, string contentType, Nullable<int> statusCode)
Microsoft.AspNetCore.Mvc.ViewFeatures.ViewExecutor.ExecuteAsync(ActionContext actionContext, IView view, ViewDataDictionary viewData, ITempDataDictionary tempData, string contentType, Nullable<int> statusCode)
Microsoft.AspNetCore.Mvc.ViewFeatures.ViewResultExecutor.ExecuteAsync(ActionContext context, ViewResult result)
Microsoft.AspNetCore.Mvc.ViewResult.ExecuteResultAsync(ActionContext context)
Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResultFilterAsync>g__Awaited|29_0<TFilter, TFilterAsync>(ResourceInvoker invoker, Task lastTask, State next, Scope scope, object state, bool isCompleted)
Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResultExecutedContextSealed context)
Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.ResultNext<TFilter, TFilterAsync>(ref State next, ref Scope scope, ref object state, ref bool isCompleted)
Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeResultFilters>g__Awaited|27_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, object state, bool isCompleted)
Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|24_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, object state, bool isCompleted)
Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(ref State next, ref Scope scope, ref object state, ref bool isCompleted)
Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, object state, bool isCompleted)
Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
SixLabors.ImageSharp.Web.Middleware.ImageSharpMiddleware.Invoke(HttpContext context)
Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware.<Invoke>g__AwaitMatch|8_1(EndpointRoutingMiddleware middleware, HttpContext httpContext, Task matchTask)
OrchardCore.Diagnostics.DiagnosticsStartupFilter+<>c__DisplayClass3_0+<<Configure>b__1>d.MoveNext() in DiagnosticsStartupFilter.cs
+
                    if (context.Response.StatusCode < 200 || context.Response.StatusCode >= 400)
Microsoft.AspNetCore.Diagnostics.StatusCodePagesMiddleware.Invoke(HttpContext context)
OrchardCore.ContentPreview.PreviewStartupFilter+<>c+<<Configure>b__1_1>d.MoveNext() in PreviewStartupFilter.cs
+
                    if (!context.Items.TryGetValue("PreviewPath", out var previewPathObject) || previewPathObject == null)
OrchardCore.Environment.Shell.Scope.ShellScope.UsingAsync(Func<ShellScope, Task> execute) in ShellScope.cs
+
                await BeforeDisposeAsync();
OrchardCore.Modules.ModularTenantContainerMiddleware.Invoke(HttpContext httpContext) in ModularTenantContainerMiddleware.cs
+
                await shellScope.UsingAsync(scope => _next.Invoke(httpContext));
Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
```

**Comment by randaratceridian** [comment]
I also just tried one more thing.

1. Opened the project in Visual Studio 2019
Changed the Nuget path to https://www.myget.org/F/orchardcore-preview/api/v3/index.json as per https://orchardcore.readthedocs.io/en/dev/docs/getting-started/templates/#adding-orchard-core-nuget-feed
Rebuilt the solution
Run the solution (starting IIS Express)

Same error
`InvalidOperationException: Could not find a resource of type 'script' named 'vendor-jquery' with version 'any'.`

**Comment by Skrypt** [comment]
Ok good we are having some progression at least. It build and you could run the setup but it fails. I will try to repro the same steps.

**Comment by Skrypt** [comment]
Ok just tested the same as above and it worked for me. Are you sure you don't have any custom theme that you also added to your solution/project? Because else it feels like if it was trying to use an unappropriate TheAgencyTheme.dll 

![image](https://user-images.githubusercontent.com/3228637/71223411-ebd67180-22a1-11ea-87e0-cafe99af6a44.png)

This is the version I get when I look at the .dll in my \bin\Debug\netcoreapp3.0\ folder.

Also, for building OC you might require to use the latest Visual Studio 2019 preview edition. It could be simply a MSBUILD failure because you would be using an incompatible version.

**Comment by deanmarcussen** [comment]
Given that this works one one of your colleagues computers and not on yours I suggest that you are experiencing problems with version resolution with NuGet @Skrypt says.

Try clearing your nuget cache folder. It is located in `/Users/.nuget`

Then when you have recomplied in visual studio check the versions of the dlls like @Skrypt has done above.
Also check the versions that have resolved in the `/Users/.nuget/packages` folder

**Comment by randaratceridian** [comment]
Hello @deanmarcussen and @Skrypt 

**_A breakthrough!_**

**Good news:**
I have an actual OrchardCore site working on my local machine! How? By following the steps here https://docs.microsoft.com/en-us/aspnet/core/tutorials/publish-to-iis?view=aspnetcore-3.1&tabs=netcore-cli and running under IIS. For whatever reason, neither `dotnet run` or IIS Express work, but good old IIS does! I can actually work on my POC!

**Additional details for fun**
You mentioned clearing nuget cache, but I had called `nuget locals all -clear` many times which clears out the folders you mentioned.

What led me down this path is I took the entire folder from my co-worker, copied it to my machine. I thought I might notice something when comparing the directories, but nothing jumped out. So if the bin folders were the same, it was not a nuget issue, which I suspected to be the case. Thinking maybe the issue was my machine the entire time, I thought IIS might work differently and *boom*, it did.

I'm sure I will need to work through some debugging issues with IIS, but this gets me started. So now, I think this ticket can actually be treated as closed. If I have other issues, I will open another ticket. This one is too long.

Thank you for your help!
